api/npm.ts

```diff
@@ -183,17 +183,35 @@ async function dependents (name: string) {
 async function typesDefinition(pkg: string, tag = 'latest') {
   let meta = await pkgJson(pkg, tag)
 
-  if (typeof meta.types === 'string' || typeof meta.typings === "string") {
+  const hasExportTypes = (meta: any) => {
+    if (typeof meta.types === 'string' || typeof meta.typings === 'string') {
+      return true
+    }
+    const hasNestedTypes = (exports: any) =>
+      typeof exports === 'object' &&
+      (typeof exports.types === 'string' ||
+      Object.values(exports)
+        .some(hasNestedTypes))
```

Comment:
These formating seems a bit confusing, how about this:
```js
const hasNestedTypes = (exports: any) => {
  if (typeof exports !== 'object') return false
  if (typeof exports.types === 'string') return true
  if (Object.values(exports).some(hasNestedTypes)) return true
  return false
}
```

Reply:
That's much better indeed!